src/prefect/server/api/concurrency_limits.py

```diff
@@ -270,7 +653,60 @@ async def decrement_concurrency_limits_v1(
         ..., description="The ID of the task run releasing the slot"
     ),
     db: PrefectDBInterface = Depends(provide_database_interface),
-) -> None:
+) -> List[MinimalConcurrencyLimitResponse]:
+    # V2: find and reconcile leases for this task run, then return current limits
+    lease_storage = get_concurrency_lease_storage()
+    lease_id = await find_lease_for_task_run(task_run_id, names, db, lease_storage)
+    if lease_id:
+        lease = await lease_storage.read_lease(lease_id)
+        if lease:
+            occupancy_seconds = (
+                datetime.now(timezone.utc) - lease.created_at
+            ).total_seconds()
+            async with db.session_context(begin_transaction=True) as session:
+                await cl_v2_models.bulk_decrement_active_slots(
+                    session=session,
+                    concurrency_limit_ids=lease.resource_ids,
+                    slots=lease.metadata.slots if lease.metadata else 0,
+                    occupancy_seconds=occupancy_seconds,
+                )
+            await lease_storage.revoke_lease(lease_id)
+
+    # Ensure legacy V1 active_slots are cleaned up as well (parity with Nebula's Redis cleanup)
+    async with db.session_context(begin_transaction=True) as session:
+        filtered_limits = (
+            await concurrency_limits.filter_concurrency_limits_for_orchestration(
+                session, tags=names
+            )
+        )
+        for cl in filtered_limits:
+            active_slots = set(cl.active_slots)
+            active_slots.discard(str(task_run_id))
+            cl.active_slots = list(active_slots)
```

Comment:
If there's a v2 limit, the v1 shouldn't exist and there shouldn't be a need for clean up.